scripts/server.py

```diff
@@ -550,66 +567,544 @@ def handle_client(conn, addr):
+                                        flush=True)
+                        else:
+                            print(
+                                f"BirdWeather Submission Error:: Detection POST - HTTP Request Exception for {soundscape_url}, Not retrying - Retries Attempted:{submission_retries} - {request_exc}",
+                                flush=True)
+                    except (
+                            requests.exceptions.JSONDecodeError,
+                            requests.exceptions.InvalidJSONError) as json_error_exec:
+                        print(
+                            f'BirdWeather Submission Error:: Detection POST - Something went wrong decoding JSON data - {json_error_exec}',
+                            flush=True)
+                    except (AttributeError, ValueError, KeyError) as value_exec:
+                        print(
+                            f'BirdWeather Submission Error:: {retry_text}Detection POST - Something went wrong accessing JSON data - {value_exec} not found',
+                            flush=True)
+                    except BaseException as dp_ex:
+                        print(
+                            f'BirdWeather Submission Error:: {retry_text}Detection POST - Something went wrong - {dp_ex}',
+                            flush=True)
+                    finally:
+                        if not can_retry:
+                            # Stop the loop as we can't be sure we can retry on this error
+                            break
+                        else:
+                            # We're retrying, retry after the calculated backoff time
+                            if request_backoff_time > 0:
+                                time.sleep(request_backoff_time)
+                                continue
+                else:
+                    print(
+                        f"BirdWeather Submission Error:: {retry_text}Detection POST did not succeed, detection_url, detection_json or soundscape_id missing or not supplied",
+                        flush=True)
+
+        else:
+            print("BirdWeather Submission Error:: Soundscape POST did not succeed, detection was not submitted",
+                  flush=True)
+
+    else:
+        print("BirdWeather Submission Error:: Soundscape POST - Soundscape URL or wav_data is missing or not supplied",
+              flush=True)
+
+
+def birdweather_submission_processor():
+    if debug_birdweather_submissions:
+        print("Starting: birdweather_submission_processor thread")
+
+    # Loop over the queue containing the data used for the BirdWeather submissions
+    while True:
+        if not bw_submission_queue.empty():
+            # Get the BirdWeather submission data, this is a dictionary containing the necessary data
+            bw_submission_data = bw_submission_queue.get()
+
+            # Verify we have data
```

Comment:
Doesn't line 1034 ensure this already?

Reply:
1304 just checks the actual if the queue has items in it, else calling .get() on a empty queue will just block. 1039 isn't really required though. removed.